src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/repository/BackupsRepository.java

```diff
@@ -0,0 +1,130 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Backups repository
+ */
+@Slf4j
+@Repository
+public class BackupsRepository {
+
+    private static final String CONFIGURATION_BACKUP_PATH = SystemProperties.getConfBackupPath();
+    private static final String BACKUP_FILE_EXTENSION = ".tar";
+    // Set maximum number of levels of directories to visit, subdirectories are excluded
+    private static final int DIR_MAX_DEPTH = 1;
+
+    /**
+     * Read backup files from configuration backup path
+     * @return
+     */
+    public List<File> getBackupFiles() {
+        try (Stream<Path> walk = Files.walk(Paths.get(CONFIGURATION_BACKUP_PATH), DIR_MAX_DEPTH)) {
+            return walk.map(Path::toString)
+                    .filter(f -> f.endsWith(BACKUP_FILE_EXTENSION))
+                    .map(this::getFile).collect(Collectors.toList());
+        } catch (IOException ioe) {
+            log.error("can't read backup files from configuration path (" + CONFIGURATION_BACKUP_PATH + ")");
+            throw new RuntimeException(ioe);
+        }
+    }
+
+    /**
+     * Get the creation date/time of a backup file
+     * @param filename
+     * @return
+     */
+    public Date getCreatedAt(String filename) {
+        Path path = getFilePath(filename);
+        try {
+            FileTime creationTime = (FileTime) Files.getAttribute(path, "creationTime");
+            return new Date(creationTime.toMillis());
+        } catch (IOException ioe) {
+            log.error("can't read backup file's creation time (" + path.toString() + ")");
+            throw new RuntimeException(ioe);
+        }
+    }
+
+    /**
+     * Delete a backup file
+     * @param filename
+     */
+    public void deleteBackupFile(String filename) {
+        Path path = getFilePath(filename);
+        try {
+            Files.deleteIfExists(path);
+        } catch (IOException ioe) {
+            log.error("can't delete backup file (" + path.toString() + ")");
+            throw new RuntimeException("deleting backup file failed");
+        }
+    }
+
+    /**
+     * Read backup file's content
+     * @param filename
+     * @return
+     */
+    public byte[] readBackupFile(String filename) {
+        Path path = getFilePath(filename);
+        try {
+            return Files.readAllBytes(path);
+        } catch (IOException ioe) {
+            log.error("can't read backup file's content (" + path.toString() + ")");
+            throw new RuntimeException(ioe);
+        }
+    }
+
+    /**
+     * Return configuration backup path with a trailing slash
+     * @return
+     */
+    public String getConfigurationBackupPath() {
+        return CONFIGURATION_BACKUP_PATH  + (CONFIGURATION_BACKUP_PATH.endsWith("/") ? "" : "/");
```

Comment:
Usually generic advice is that these should use `File.separator` instead of `"/"`. There is some limited use of File.separator in the codebase already.

DefaultFilepaths.CONF_BACKUP_PATH uses hard coded "/" but maybe there is some tiny change that someone will run this with overridden CONF_BACKUP_PATH on a system with different file separator.

I don't see how File.separator would break otherwise working configuration, so there should not be too much of a downside for changing. But this is not a super important detail. It's more of a style issue.

Reply:
You're right, the default configuration includes the trailing `/`. Since the default value can be overridden, it's possible for an admin to change the location and set the new value without the trailing `/`. Maybe not very common, but possible. However, I replaced the hardcoded `/` with `File.separator`.